src/status_im/data_store/user_statuses.cljs

```diff
@@ -40,7 +40,14 @@
 
 (defn save-statuses-tx
   "Returns tx function for saving message user statuses"
-  [user-statuses]
-  (fn [realm]
-    (doseq [user-status user-statuses]
-      ((save-status-tx user-status) realm))))
+  ([user-statuses]
+   (save-statuses-tx user-statuses nil))
+  ([user-statuses on-success]
+   (fn [realm]
+     (let [success (reduce *
```

Comment:
1. [`doseq`](https://clojuredocs.org/clojure.core/doseq) always returns nil, not a collection. It is meant for side effects. So here it could be replaced with `for`. It is literally `(reduce * [])` now which always returns `1` as `(*)` returns `1` `¯\_(ツ)_/¯`
2. `(reduce * [true false true ...])` even if collection contains only one `false` value the whole collection of messages will be treated as not persisted, do we really want this? 
3. this code is not going to be run in the app currently as `seen` messages are not working. How soon are we going to "ressurect" them? If we don't have a plan to do this better just to remove dead code.

Reply:
same here, probably we need to wait for the transaction to be committed before dispatching